app/package/controllers.py

```diff
@@ -440,12 +441,14 @@ def get_metadata(publisher, package):
             return handle_error('DATA_NOT_FOUND',
                                 'No metadata found for the package',
                                 404)
+        if not isinstance(data.descriptor, dict):
+            raise Exception("descriptor should be type of dict")
```

Comment:
I can't understand how it could become non-dict? It is generally better to put such validation check on input (write), and not on read.